lib/compress.js

```diff
@@ -1724,12 +1734,151 @@ merge(Compressor.prototype, {
+                operator: "||",
+                left: this.condition,
+                right: alternative
+            }) : this.condition;
+            if (!alternative) return make_node(AST_Binary, this, {
+                operator: "&&",
+                left: this.condition,
+                right: consequent
+            });
+            var node = this.clone();
+            this.consequent = consequent;
+            this.alternative = alternative;
+            return node;
+        });
+        def(AST_Unary, function(compressor, first_in_statement){
+            switch (this.operator) {
+              case "delete":
+              case "++":
+              case "--":
+                return this;
+              case "typeof":
+                if (this.expression instanceof AST_SymbolRef) return null;
+              default:
+                if (first_in_statement && is_iife_call(this.expression)) return this;
+                return this.expression.drop_value(compressor, first_in_statement);
+            }
+        });
+        def(AST_SymbolRef, function() {
+            return this.undeclared() ? this : null;
+        });
+        def(AST_Object, function(compressor, first_in_statement){
+            var values = trim(this.properties, compressor, first_in_statement);
+            return values && AST_Seq.from_array(values);
+        });
+        def(AST_ObjectProperty, function(compressor, first_in_statement){
+            return this.value.drop_value(compressor, first_in_statement);
+        });
+        def(AST_Array, function(compressor, first_in_statement){
+            var values = trim(this.elements, compressor, first_in_statement);
+            return values && AST_Seq.from_array(values);
+        });
+        def(AST_Dot, function(compressor, first_in_statement){
+            if (!compressor.option("pure_getters")) return this;
+            return this.expression.drop_value(compressor, first_in_statement);
+        });
+        def(AST_Sub, function(compressor, first_in_statement){
+            if (!compressor.option("pure_getters")) return this;
+            var expression = this.expression.drop_value(compressor, first_in_statement);
+            if (!expression) return this.property.drop_value(compressor, first_in_statement);
+            var property = this.property.drop_value(compressor);
+            if (!property) return expression;
+            return make_node(AST_Seq, this, {
+                car: expression,
+                cdr: property
+            });
+        });
+        def(AST_Seq, function(compressor){
+            var cdr = this.cdr.drop_value(compressor);
+            if (cdr === this.cdr) return this;
+            if (!cdr) return this.car;
+            return make_node(AST_Seq, this, {
+                car: this.car,
+                cdr: cdr
+            });
+        });
+    })(function(node, func){
+        node.DEFMETHOD("drop_value", func);
```

Comment:
Would it bother you to change the name of the function to something more descriptive like `drop_irrelevant` or `drop_side_effect_free`?